File: Desktop/Desktop.Data/PrecioDia.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Desktop.Data
{
    public class PrecioDia
    {
        private string _url;
        public int ID { get; set; }
        public DateTime Fecha { get; set; }
        public int Precio { get; set; }
        public int HabitacionID { get; set; }
        public Habitacion Habitacion { get; set; }

        public PrecioDia()
        {
            _url = "http://127.0.0.1:3000/";
            GetHelper<PrecioDia>.Url = _url;
            Conexion.Cliente.BaseAddress = new Uri(_url);
        }

        public bool CrearPrecioDia(PrecioDia precioDia)
        {
            Conexion.Url = _url;
            Conexion.Cliente.BaseAddress = new Uri(_url);
            var queryParams = new Dictionary<string, string>
            {
                { "fecha", precioDia.Fecha.ToString() },
                { "precio", precioDia.Precio.ToString() },
                { "habitacionId", precioDia.HabitacionID.ToString() },
            };
            return Conexion.Post(queryParams, "crearPrecioDia");
        }

        public bool ActualizarPrecioDia(PrecioDia precioDia)
        {
            Conexion.Url = _url;
            Conexion.Cliente.BaseAddress = new Uri(_url);
            var queryParams = new Dictionary<string, string>
            {
                { "id", precioDia.ID.ToString() },
                { "fecha", precioDia.Fecha.ToString() },
                { "precio", precioDia.Precio.ToString() },
                { "habitacionId", precioDia.HabitacionID.ToString() },
            };
            return Conexion.Post(queryParams, "actualizarPrecioDia");
        }

        public List<PrecioDia> ObtenerPrecioDiaPorFecha(DateTime fecha)
        {
            GetHelper<PrecioDia>.Url = _url;
            var queryParams = new Dictionary<string, string>
            {
                { "fecha", fecha.ToString() },
            };
            return GetHelper<PrecioDia>.GetListPorId(queryParams, "obtenerPrecioDiaPorFecha");
        }
        public List<PrecioDia> ObtenerPrecioDiaPorFechaYHabitacionId(DateTime fecha, int habitacionId)
        {
            GetHelper<PrecioDia>.Url = _url;
            var queryParams = new Dictionary<string, string>
            {
                { "fecha", fecha.ToString() },
                { "habitacionId", habitacionId.ToString() },
            };
            return GetHelper<PrecioDia>.GetListPorId(queryParams, "obtenerPrecioDiaPorFechaYHabitacionId");
        }
    }
}

File: server/bussines/precioDia.js
const precioDiaDao = require("../dao/precioDiaDao.js");

const crearPrecioDia = function(req, res){
    let precioDia = 
    {
        fecha:  req.body.fecha,
        precio: parseInt(req.body.precio),
        habitacionId: parseInt(req.body.habitacionId)
    };
    precioDiaDao.crearPrecioDia(precioDia).then((success) => {
        res.send(success.toString());
    }).catch((error) => {
        console.log(error);
    });
};

const actualizarPrecioDia = function(req, res){
    let precioDia = 
    {
        id:     parseInt(req.body.id),
        fecha:  req.body.fecha,
        precio: parseInt(req.body.precio),
        habitacionId: parseInt(req.body.habitacionId)
    };
    precioDiaDao.actualizarPrecioDia(precioDia).then((success) => {
        res.send(success.toString());
    }).catch((error) => {
        console.log(error);
    });
};

const obtenerPrecioDiaPorFechaYHabitacionId = function(req, res){
    let precioDia = 
    {
        fecha:  req.body.fecha,
        habitacionId: parseInt(req.body.habitacionId)
    };
    precioDiaDao.obtenerPrecioDiaPorFechaYHabitacionId(precioDia).then((success) => {
        res.send(success);
    }).catch((error) => {
        console.log(error);
    });
};

const obtenerPrecioDiaPorFecha = function(req, res){
    let precioDia = 
    {
        fecha:  req.body.fecha
    };
    precioDiaDao.obtenerPrecioDiaPorFecha(precioDia).then((success) => {
        res.send(success);
    }).catch((error) => {
        console.log(error);
    });
};

module.exports.crearPrecioDia = crearPrecioDia;
module.exports.actualizarPrecioDia = actualizarPrecioDia;
module.exports.obtenerPrecioDiaPorFecha = obtenerPrecioDiaPorFecha;
module.exports.obtenerPrecioDiaPorFechaYHabitacionId = obtenerPrecioDiaPorFechaYHabitacionId;